okcupyd/json_search.py

```diff
@@ -70,15 +75,59 @@ def fetch_once(self, count=18):
 
 class SearchJSONFetcher(object):
 
-    def __init__(self, session=None, **options):
+    def __init__(self, session=None, order_by="MATCH", location=None, keywords=None, **options):
         self._session = session or Session.login()
-        self._options = options
+        self._order_by = order_by.upper()
+        if self._order_by not in ('MATCH', 'ONLINE', 'SPECIAL_BLEND'): # TODO: haven't verified that options other than 'MATCH' still work; the web interface no longer provides a way to modify this
+            raise TypeError
+        self._location = location # location string, haven't implemented
+        self._keywords = keywords # list or space-delimited string of words to match, haven't implemented
         self._parameters = search_filters.build(**options)
 
     def _query_params(self, after=None, count=None):
         search_parameters = {
             'after': after,
-            'limit': count
+            'limit': count,
+            # other admin parameters
+            'debug': 0,
+            'username': "",
+            'save_search': 1,
+            'order_by': self._order_by,
+
+            'i_want': None,
+            'they_want': None,
+            'tagOrder': [],
+
+            # optional parameters
+            'minimum_height': None,
+            'maximum_height': None,
+            'languages': 0,
+            'speaks_my_language': 0, # TODO shall we make 1 the default?
+            'ethnicity': [],
```

Comment:
I agree it'd be useful to have some way to specify in filters that they should push a default value when not invoked. (But it should be optional, as we may only want this for some output keys.) I don't get how your EDIT would work. Are you imagining having the filters be invoked even if there is no matching key? Would the architecture need to be changed to specify what default values should be passed to the `transform` function? I'm not thrilled about this prospect, because then most of the filters would need to use that `always` decider, and then would have to reproduce the previous `decide` logic inside their own implementation to manage how to proceed. That's for the filters which have non-trivial `decide` logic already, which is a few of them, for example the height filters and all the gender/orientation ones.

Adding the defaults to part of the base request dict seemed to me the easiest implementation. Granted it does separate the declaration of whether a default value is needed in the request, and where it needs to be, from the filter definition.